tests/experimental/evals/functions/test_executor.py

```diff
@@ -0,0 +1,239 @@
+import asyncio
+import signal
+from unittest.mock import AsyncMock, Mock
+
+import nest_asyncio
+from phoenix.experimental.evals.functions.executor import (
+    AsyncExecutor,
+    SyncExecutor,
+    get_executor_on_sync_context,
+)
+
+# AsyncExecutor tests
+
+
+async def test_async_executor_executes():
+    async def dummy_fn(payload: int) -> int:
+        return payload - 1
+
+    executor = AsyncExecutor(dummy_fn, concurrency=10, max_retries=0)
+    inputs = [1, 2, 3, 4, 5]
+    outputs = await executor.execute(inputs)
+    assert outputs == [0, 1, 2, 3, 4]
+
+
+async def test_async_executor_executes_many_tasks():
+    async def dummy_fn(payload: int) -> int:
+        return payload
+
+    executor = AsyncExecutor(dummy_fn, concurrency=10, max_retries=0)
+    inputs = [x for x in range(100)]
+    outputs = await executor.execute(inputs)
+    assert outputs == inputs
+
+
+def test_async_executor_runs_synchronously():
+    async def dummy_fn(payload: int) -> int:
+        return payload - 2
+
+    executor = AsyncExecutor(dummy_fn, concurrency=10, max_retries=0)
+    inputs = [1, 2, 3, 4, 5]
+    outputs = executor.run(inputs)
+    assert outputs == [-1, 0, 1, 2, 3]
+
+
+async def test_async_executor_execute_exits_early_on_error():
+    async def dummy_fn(payload: int) -> int:
+        if payload == 3:
+            raise ValueError("test error")
+        return payload - 1
+
+    executor = AsyncExecutor(
+        dummy_fn, concurrency=1, max_retries=0, exit_on_error=True, fallback_return_value=52
+    )
+    inputs = [1, 2, 3, 4, 5]
+    outputs = await executor.execute(inputs)
+    assert outputs == [0, 1, 52, 52, 52]
+
+
+def test_async_executor_run_exits_early_on_error():
+    async def dummy_fn(payload: int) -> int:
+        if payload == 3:
+            raise ValueError("test error")
+        return payload - 1
+
+    executor = AsyncExecutor(
+        dummy_fn, concurrency=1, max_retries=0, exit_on_error=True, fallback_return_value=52
+    )
+    inputs = [1, 2, 3, 4, 5]
+    outputs = executor.run(inputs)
+    assert outputs == [0, 1, 52, 52, 52]
+
+
+async def test_async_executor_can_continue_on_error():
+    async def dummy_fn(payload: int) -> int:
+        if payload == 3:
+            raise ValueError("test error")
+        return payload - 1
+
+    executor = AsyncExecutor(
+        dummy_fn, concurrency=1, max_retries=0, exit_on_error=False, fallback_return_value=52
+    )
+    inputs = [1, 2, 3, 4, 5]
+    outputs = await executor.execute(inputs)
+    assert outputs == [0, 1, 52, 3, 4]
+
+
+async def test_async_executor_sigint_handling():
+    import os
+    import time
+
+    class InterruptingIterator:
```

Comment:
Previously, this test was simulating a `SIGINT` by calling the signal handler directly on the `Executor` class. Since this signal handler is no longer explicitly held by the class to prevent associating primitives with the wrong event loop, we are adjusting our testings strategy. The executor is passed a mined payload that will send a SIGINT to the process that's running the executor when processing the nth element.